pkg/plugin/admissioncontrollers/webhook/admission_controller.go

```diff
@@ -199,11 +218,10 @@ func (c *admissionController) processPod(req *admissionv1.AdmissionRequest) *adm
 	return admissionResponseBuilder(uid, true, "", patchBytes)
 }
 
-func (c *admissionController) processWorkload(req *admissionv1.AdmissionRequest) *admissionv1.AdmissionResponse {
+func (c *admissionController) processWorkload(req *admissionv1.AdmissionRequest, namespace string) *admissionv1.AdmissionResponse {
```

Comment:
The `namespace` parameter is unused here.

Reply:
Also, I think we need to exclude workloads the same way we do pods, by checking the namespace against (`process/bypass)Namespaces` and `(label/noLabel)Namespaces`. Otherwise we're affecting workloads that will not be scheduled by YuniKorn.

Reply:
Could you double check this?

If something is in `noLabelNamespaces`, we still add the `scheduler` label, we skip `disableStateAware`, `queue` and `applicationId`. But to me it looks like Yunikorn will process it.

Reply:
Yunikorn processes it, however because the applicationId is missing the scheduler plug-in goes into no-op mode and passes it through to the underlying default scheduler. 

In standard mode, Yunikorn just bails out and won't schedule the pod at all. ☹️ 

Technically even this check isn't quite good enough, because we don't account for user-supplied applicationId being present. We should probably check for existing appId or spark appId and opt back in to the security checks. 

In that case our logic becomes:

shouldProcess && (appId/sparkAppId present || shouldLabel)

That's ugly, but I think correct. Probably build a helper function for this and we can call it from wherever needed.

Reply:
Done.

I had to add extra code to extract the labels from deployments, replicasets, etc. This required some refactoring and new tests.